### ISSUE TITLE
No/delayed notification after long inactivity on Android 7

### ISSUE THREAD

**Issue by willempx** [Telegram-FOSS-Team/Telegram-FOSS#338]
**Environment**:
- Lineage Android 7.1.2
- Telegram-FOSS 5.7.1 from f-droid
- No GApps/googlestore installed so no GCM
- Battery optimizations for Telegram-FOSS is disabled

**Repreducibility**:
Probably always.

**Steps**:
1. Open the homescreen of the phone (to make sure Telegram is not in foreground).
2. Lock the phone (i.e. press power button).
3. Wait a long time (e.g. 2 hours, not just 7 minutes) during which the phone is not used and no one sends a message to the phone. This step is important.
4. On another device, send a Telegram message to the phone.

**Actual result**:
After step 4, there is no notification, i.e. no notification LED turns on and no notification sound is played (for the first half hour, I did not test for longer)
    
**Expected result**:
After step 4, it takes at most 5 seconds until notification LED and sound.

**Comment by sobrus** [comment]
I have the same issue on Android 6.0 (MIUI 10). I have disabled battery optimizations, disabled doze mode entirely, frozen xiaomi battery saving app, but still notifications are unreliable.
That's why I'm sticking with official GCM version via microg. It would be great to have GCM option available. 
On the other hand, Telegram FOSS works great on my older Android 4.3 phone.

**Comment by Younes-L** [comment]
@sobrus clearly the issue is your MIUI/ROM rather than tg-foss
If you have a "lock in memory" 🔒 button in multitasking view (overview), try enabling it for tg-foss to prevent it from being killed
Also, isn't MIUI 10 based on android 9 ??

**Comment by sobrus** [comment]
No, It's locked and it doesn't get killed for sure (4GB RAM + zram with swappiness set at 100). The issue was present on MIUI 8 as well. Xiaomi is messing Android pretty badly anyway, apps get frozen, killed, memory management is awkward and even doze mode disables GPS (even for whitelisted apps of course). 

Android version is device-dependend. Redmi Note 4 has only Android 6.0 (probably Xiaomi didn't want to pay Mediatek for system updates).

**Comment by mzakharo** [comment]
I am also experiencing arbitrary long-delays (sometimes more than 5 minutes) when receiving messages on my non-gapps Lineage phone, while my other stock Samsung G-Apps enabled device receives them virtually instantly. 

My phone is running  Official Lineage 18.1 (Android 11),  No-gapps,   Sony XZ2c.   Telegram-FOSS Battery Opt disabled.

Does anyone know if there is something that can be done to further debug/resolve this issue?
